### DISCUSSION TITLE
Can I redefine existing utility classes?

### DISCUSSION THREAD

**Discussion by namad** [tailwindlabs/tailwindcss#12339] [Help] [answered]
What is the best practice to modify existing utility classes? I made my custom plugin for `bg` and `border` to use variables

```
plugin(function ({ matchUtilities, theme }) {
    matchUtilities(
        {
            border: (value) => ({
                '--border-color': value,
                'border-color': 'var(--border-color)'
            }),
        },
        {
            values: theme('colors')
        }
    )
})

```
and now I got two declarations in my CSS

this is an output from my plugin

```
.border-stroke-base-200 {
    --border-color: var(--stroke-base-200);
    border-color: var(--border-color);
}
```

and then there is a native one that is being overriden
```
.border-stroke-base-200 {
    border-color: var(--stroke-base-200);
}
```

Is there a way to reduce it only to have mine instead?

The reason I am doing it is to have better way to set transparency to my colours, so I added another plugin that takes `--border-color` and use `color-mix()` to fade it out

```
.border-fade-80 {
    border-color: color-mix(in srgb, var(--border-color), transparent calc(100% - 80%));
}
```

**Comment by wongjn** [comment, marked as answer]
> Is there a way to reduce it only to have mine instead?

Yes there is. [Disable the core plugin](https://tailwindcss.com/docs/theme#disabling-an-entire-core-plugin), like:

```js
/** @type {import('tailwindcss').Config} */
module.exports = {
  corePlugins: {
    borderColor: false,
  }
}
```

**Reply by namad** [reply]
Ace! Appreciate it!
